### PR TITLE
Moving to Python3

### PR DESCRIPTION
**Pull request by iambibhas**
Related to https://github.com/hasgeek/hgapp/issues/39

### CODE REVIEW

File: funnel/models/project.py [hunk 1]
```diff
@@ -324,7 +324,7 @@ def datelocation(self):
                 end_date=schedule_end_at_date.strftime("%d %b"),
                 year=schedule_end_at_date.year,
             )
-        return u', '.join(filter(None, [daterange, self.location]))
+        return ', '.join([_f for _f in [daterange, self.location] if _f])
```
Comment (jace) [inline funnel/models/project.py]: Is this an accurate replacement?

File: funnel/models/proposal.py [hunk 2]
```diff
@@ -532,7 +532,7 @@ def url_id_name(self):
         likely never be called directly on an instance. It exists for the SQL
         comparator that will be called to load the instance.
         """
-        return unicode(self.url_id)
+        return self.url_id
```
Comment (jace) [inline funnel/models/proposal.py]: `return str(self.url_id)` as it's an integer.

File: funnel/views/proposal.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def view(self):
         delcommentform = DeleteCommentForm()
 
         links = [
-            Markup(linkify(unicode(escape(l))))
+            Markup(linkify(escape(l)))
```
Comment (jace) [inline funnel/views/proposal.py]: Needs to be `str()` as `escape` returns a HTML string.

File: funnel/views/schedule.py [hunk 4]
```diff
@@ -34,53 +34,63 @@
...
+        + list(
+            dict(
+                {'modal_url': session.url_for(with_modal_url)} if with_modal_url else {}
+            ).items()
+        )
+        + list(
+            dict(
+                {'delete_url': session.url_for('delete')} if with_delete_url else {}
+            ).items()
+        )
```
Comment (jace) [inline funnel/views/schedule.py]: This ugly block should be replaced with a single dict and two update operations.

File: funnel/models/project.py (hunk 1 above)
Comment (iambibhas) [reply funnel/models/project.py]: Seems so.

**Pull request merged**